src/Tezos/AddressChecksum.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "AddressChecksum.h"
+
+#include "../Hash.h"
+#include "../HexCoding.h"
+#include <cctype>
+
+using namespace TW;
+using namespace TW::Tezos;
+
+std::string Tezos::checksumed(const Address& address) {
+  // TODO: implement checksum
```

Comment:
Calculate a checksum by taking your prefixed data (tz1 + <whatever>), sha256 hashing it twice, then taking the first four bytes. 

Append this to the end of the data (so you have tz1 + <whatever> + <checksum>) and base58encode it